main/ZgatewayBT.ino

```diff
@@ -223,7 +223,7 @@ void MiJiaDiscovery(char* mac) {
       //component type,name,availability topic,device class,value template,payload on, payload off, unit of measurement
   };
 
-  createDiscoveryFromList(mac, MiJiasensor, MiJiaparametersCount);
+  createDiscoveryFromList(mac, MiJiasensor, MiJiaparametersCount, "", "", "");
```

Comment:
You can find all the model reference below:
https://github.com/1technophile/OpenMQTTGateway/blob/e17b5b9c9d97ac83ef51aefdf8c93de408d8046e/main/ZgatewayBT.ino#L836

Reply:
What about add p->sensorModel in each *Discovery functions ? to avoid to hardcoded 2 times the Model ? and also to be able to handle when 2 models use the same Discovery function...

Reply:
Good catch!

Reply:
and same think for p->manufacturer ?

Reply:
this one doesn't exist in the structure

Reply:
Hum, I saw the manufacturer in the logs when it scans the devices. I guess it's not available for all devices ?

Reply:
Sorry I meant p->name for the name ?

Reply:
@1technophile 
This line
```
void CLEARGRASSTRHKPADiscovery(vector<BLEdevice>::iterator p) {}
```
Doesn't work ...
I get
```
ZgatewayBT:282:32: error: variable or field 'CLEARGRASSTRHKPADiscovery' declared void
 void CLEARGRASSTRHKPADiscovery(vector<BLEdevice>::iterator p) {

ZgatewayBT:282:32: error: 'vector' was not declared in this scope
...
ZgatewayBT:282:39: error: 'BLEdevice' was not declared in this scope
 void CLEARGRASSTRHKPADiscovery(vector<BLEdevice>::iterator p) {
                                       ^
ZgatewayBT:282:49: error: '::iterator' has not been declared
 void CLEARGRASSTRHKPADiscovery(vector<BLEdevice>::iterator p) {
```
How can I fix that ?

Reply:
>Sorry I meant p->name for the name ?

It will not be p->name, you would need to retrieve it from the BLEjson object. After checking it exists.

Reply:
You need to pass the sensor model into the method:
 CLEARGRASSTRHKPADiscovery((char*)macWOdots.c_str(), p->sensorModel);

Reply:
Yes, it seems hard to get the name, So I will hardcode it for now

Reply:
done